test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -0,0 +1,246 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/protoToYul.h>
+#include <test/tools/ossfuzz/yulProto.pb.h>
+
+#include <ostream>
+#include <sstream>
+
+using namespace std;
+
+namespace yul
+{
+namespace test
+{
+namespace yul_fuzzer
+{
+ostream& operator<<(ostream& _os, BinaryOp const& _x);
+ostream& operator<<(ostream& _os, Block const& _x);
+
+ostream& operator<<(ostream& _os, Literal const& _x)
+{
+	return _os << "(" << _x.val() << ")";
+}
+
+ostream& operator<<(ostream& _os , VarRef const& _x)
+{
+	return _os  << "x_" << (static_cast<uint32_t>(_x.varnum()) % 10);
+}
+
+ostream& operator<<(ostream& _os , Expression const& _x)
+{
+	if (_x.has_varref())
+		return _os  << _x.varref();
+	else if (_x.has_cons())
+		return _os  << _x.cons();
+	else if (_x.has_binop())
+		return _os  << _x.binop();
+	else
+		return _os  << "1";
+}
+
+ostream& operator<<(ostream& _os , BinaryOp const& _x)
+{
+	switch (_x.op()) {
+		case BinaryOp::ADD:
+			_os << "add(";
+			break;
+		case BinaryOp::SUB:
+			_os << "sub(";
+			break;
+		case BinaryOp::MUL:
+			_os << "mul(";
+			break;
+		case BinaryOp::DIV:
+			_os << "div(";
+			break;
+		case BinaryOp::MOD:
+			_os << "mod(";
+			break;
+		case BinaryOp::XOR:
+			_os << "xor(";
+			break;
+		case BinaryOp::AND:
+			_os << "and(";
+			break;
+		case BinaryOp::OR:
+			_os << "or(";
+			break;
+		case BinaryOp::EQ:
+			_os << "eq(";
+			break;
+		case BinaryOp::LT:
+			_os << "lt(";
+			break;
+		case BinaryOp::GT:
+			_os << "gt(";
+			break;
+	}
+	return _os << _x.left() << "," << _x.right() << ")";
+}
+
+// New var numbering starts from x_10 until x_16
+ostream& operator<<(ostream& _os , VarDecl const& _x)
+{
+	_os << "let x_" << ((_x.id() % 7) + 10) << " := " << _x.expr() << "\n";
+	return _os;
+}
+
+ostream& operator<<(ostream& _os , TypedVarDecl const& _x)
+{
+	_os << "let x_" << ((_x.id() % 7) + 10);
+	switch (_x.type())
+	{
+		case TypedVarDecl::BOOL:
+			_os << ": bool := " << _x.expr() << " : bool\n";
```

Comment:
The `:bool` usually does not follow every expression, just if it is a literal.

Reply:
okay, at the moment typed statements are disabled. will incorporate this once I include typed statements into the mix.